test/js/bun/http/async-iterator-stream.test.ts

```diff
@@ -0,0 +1,195 @@
+import { describe, expect, test } from "bun:test";
+import { bunExe, bunEnv } from "harness";
+import path from "path";
+
+describe("Streaming body via", () => {
+  test("async generator function", async () => {
+    const server = Bun.serve({
+      port: 0,
+
+      async fetch(req) {
+        return new Response(async function* yo() {
+          yield "Hello, ";
+          await Bun.sleep(30);
+          yield Buffer.from("world!");
+          return "!";
+        });
+      },
+    });
+
+    const res = await fetch(`${server.url}/`);
+    const chunks = [];
+    for await (const chunk of res.body) {
+      chunks.push(chunk);
+    }
+
+    expect(Buffer.concat(chunks).toString()).toBe("Hello, world!!");
+    expect(chunks).toHaveLength(2);
+    server.stop(true);
+  });
+
+  test("async generator function throws an error but continues to send the headers", async () => {
+    const server = Bun.serve({
+      port: 0,
+
+      async fetch(req) {
+        return new Response(
+          async function* () {
+            throw new Error("Oops");
```

Comment:
Can we either move this test to a subprocess or add a special property that hides the errors from being printed? This makes our GitHub CI spammy with errors

Reply:
That should be fixed in the GitHub actions printer IMO

Reply:
Related to https://github.com/oven-sh/bun/pull/8949